### ISSUE TITLE
Missing CS8166 when accessing [UnscopedRef] property to assing to an unscoped local

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#63526]
**Version Used**:  4.4.0-2.22419.1 (ae566380)

**Steps to Reproduce**:

1. Paste the following code:
```csharp
ref struct A
{
    [UnscopedRef]
    public Span<byte> Span => default;
}

class Foo
{
    void Bar(ref A a)
    {
        Span<byte> span = a.Span;
    }
}
```

**Expected Behavior**:

The code should produce a CS8166 error (same as #63524, but correctly here), because `span` is not `scoped`.

**Actual Behavior**:

Code compiles fine.

![](https://user-images.githubusercontent.com/10199417/186033466-46973d7b-cf8e-4da6-9083-40802aa121a4.png)

cc. @cston

**Comment by jaredpar** [comment]
The lifetime of a local is inferred at declaration based on the initializer expression (as it has since the initial `ref` safety work done in C# 7.2). In the case the RHS is not safe to return then the local is implicitly marked as `scoped`. That means the following two declarations are equivalent: 

```csharp
Span<byte> span1 = stackalloc int[42]; 
scoped Span<byte> span2 = stackalloc int[42]; 
```

So in this case the behavior is expected as the compiler silently infers the declaration is `scoped`.

**Comment by cston** [comment]
Reopening. We should report a similar error to the error reported in #63524 as @Sergio0694 indicated in the description.
